webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/TechniqueVersion.scala

```diff
@@ -37,290 +37,65 @@
-            "the characters . + - : ~ (full stop, plus, hyphen, colon, tilde).")
-      case _ => ParseVersion.parse(value) match {
-        case Right(v)  => v
-        case Left(err) => throw new TechniqueVersionFormatException(err)
+  /*
+   * A technique version is *much* simpler than a full blown version.
+   * We can only have: a.b.c.etc+commitId
+   *
+   * NOTE: it seems that in existing env, sometimes epoch was written when it should
+   * not have. So we must accept also leading number+":".
+   */
+  def authorizedChar = """([0-9]:)*[.0-9]+""".r.pattern
+
+  def parse(value: String): Either[String, TechniqueVersion] = {
+    val (v, revId) = {
+      val parts = value.split("\\+")
+      if(parts.size == 1) {
+        (value, None)
+      } else {
+        (parts.take(parts.size-1).mkString("+"), Some(RevId(parts(parts.size-1).trim)))
       }
     }
-  }
-
-  def compare(upsreamTechniqueVersion: UpstreamTechniqueVersion): Int = {
-    parsed.compareTo(upsreamTechniqueVersion.parsed)
-  }
-
-}
-
-
-// alternative implementation with a parser
-
-/*
- * A version is composed of :
- * - an optionnal Epoch: a natural number followed by ":"
- * - a list of version parts seperated by "." or "-" or "~"
- * - a change from letters to digits define a version part as if
- *   separated with a "-", "1.2.2sp1" == "1.2.2-sp1"
- * - "." is higher than "-" so that "1.2.2" > "1.2-2"
- * - "~" modify the following part to mean "before what precede"
- * - some words also mean "before", they are, sorted in that order:
- *   snapshot[s], nightly|nightlies, alpha, beta, milestone, rc
- *   so that "1.2-alpha1" == "1.2~alpha-1"
- * - all other words mean "after", and they are sorted alphabetically
- */
-final case class Version(epoch: Long, head: PartType, parts: List[VersionPart]) extends ToVersionString with Ordered[Version] {
-  def toVersionString: String =
-    (if(epoch == 0) "" else epoch.toString + ":") + toVersionStringNoEpoch
-
-  def toVersionStringNoEpoch = head.toVersionString + parts.map(_.toVersionString).mkString("")
-
-  override def compare(other: Version): Int = Version.compare(this, other)
-}
-
-object Version {
-  //create a list of default value, ie a list of ".0", of the given size
-  def defaultList(size: Int) = List.fill(size)(VersionPart.After(Separator.Dot, PartType.Numeric(0)))
-  @scala.annotation.tailrec
-  def compareList(a: List[VersionPart], b: List[VersionPart]): Int = {
-    (a, b) match {
-      case (Nil, Nil) => 0
-      case (Nil, l  ) => compareList(defaultList(l.size), l)
-      case (l  , Nil) => compareList(l, defaultList(l.size))
-      case (h1::t1, h2::t2) =>
-        val h = VersionPart.compare(h1, h2)
-        if(h == 0) compareList(t1, t2) else h
+    if(authorizedChar.matcher(v).matches()) {
+      ParseVersion.parse(value).map(x => TechniqueVersion(x, revId))
+    } else {
+      Left(s"Error: technique version can only have [.0-9] chars in main part + revisionId, not: '${value}'")
```

Comment:
Don't you want a PureResult[TechniqueVersion] here ?

Reply:
with an error type instead of a string?

Reply:
Having just a string make integration with third parties easier, but for consistancy it should be